cmd/nilaway/main.go

```diff
@@ -18,10 +18,105 @@
 package main
 
 import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
 	"go.uber.org/nilaway"
+	"go.uber.org/nilaway/config"
+	"golang.org/x/tools/go/analysis"
 	"golang.org/x/tools/go/analysis/singlechecker"
 )
 
+// Analyzer is identical to the one in nilaway.go, except that it overrides the run function for
+// extra filtering of errors, since the singlechecker does not support error suppression like other
+// popular linter drivers.
+var Analyzer = &analysis.Analyzer{
+	Name:       nilaway.Analyzer.Name,
+	Doc:        nilaway.Analyzer.Doc,
+	Run:        run,
+	FactTypes:  nilaway.Analyzer.FactTypes,
+	ResultType: nilaway.Analyzer.ResultType,
+	Requires:   nilaway.Analyzer.Requires,
+}
+
+var (
+	// _includeErrorsInFiles is a driver flag for specifying the list of file prefixes to only report errors.
+	_includeErrorsInFiles string
+	// _excludeErrorsInFiles is a driver flag for specifying the list of file prefixes to not report errors.
+	_excludeErrorsInFiles string
+)
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	// NilAway by default analyzes all packages, including dependencies. Even if specified to
+	// exclude packages from analysis via configurations, NilAway can still report errors on
+	// packages that are not analyzed if the nilness flow happens within the analyzed package, but
+	// the flow concerns a struct that is in an excluded package. The usual way to handle them is
+	// to suppress them at the driver level, but singlechecker does not support that yet. Therefore,
+	// here we add extra logic to filter the errors.
+
+	// Properly parse the error suppression flags.
+	var files [2][]string
+	for i, flagVal := range [...]string{_includeErrorsInFiles, _excludeErrorsInFiles} {
+		if flagVal == "" {
+			files[i] = nil
+			continue
+		}
+
+		// Convert the file paths to absolute paths.
+		list := strings.Split(flagVal, ",")
+		for i := range list {
+			p, err := filepath.Abs(list[i])
+			if err != nil {
+				return nil, fmt.Errorf("incorrect file path: %s", list[i])
```

Comment:
Do we not want to include the underlying error `err` here somehow? Just to get a more specific error message.